src/main/java/org/jenkinsci/plugins/workflow/steps/scm/SCMStep.java

```diff
@@ -24,32 +24,30 @@
 
 package org.jenkinsci.plugins.workflow.steps.scm;
 
+import com.google.common.collect.ImmutableSet;
 import hudson.FilePath;
 import hudson.Launcher;
 import hudson.model.Run;
 import hudson.model.TaskListener;
 import hudson.model.listeners.SCMListener;
 import hudson.scm.SCM;
 import hudson.scm.SCMRevisionState;
-
 import java.io.File;
-import java.io.Serializable;
 import java.util.Map;
+import java.util.Set;
 import java.util.TreeMap;
-
 import javax.annotation.Nonnull;
-import javax.inject.Inject;
-
-import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
-import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
-import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
-import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution;
 import org.kohsuke.stapler.DataBoundSetter;
 
 /**
  * A step which uses some kind of {@link SCM}.
  */
-public abstract class SCMStep extends AbstractStepImpl implements Serializable {
+public abstract class SCMStep extends Step {
```

Comment:
`Serializable` seems to be left over from https://github.com/jenkinsci/pipeline-plugin/commit/f6d57d37cbfaaca2d290c3d78f52ec6d60fa29b0 when the `StepExecution` was an inner class.